src/catalog/src/durable/transaction.rs

```diff
@@ -109,7 +109,7 @@ impl<'a> Transaction<'a> {
                 a.database_id == b.database_id && a.name == b.name
             })?,
             items: TableTransaction::new(items, |a: &ItemValue, b| {
-                a.schema_id == b.schema_id && a.name == b.name
+                a.schema_id == b.schema_id && a.name == b.name && a.is_type() == b.is_type()
```

Comment:
Wouldn't this allow for example a table and type to exist in the same schema with the same name, which shouldn't be allowed?

Reply:
Adjusted this logic accordingly; PTAL!